public/src/inverted-index.js

```diff
@@ -14,49 +15,52 @@ class InvertedIndex {
     this.search_terms = [];
   }
   /**
+   * Create `index
+   *
    * creates an inverted index from the specified fileName or JsonObject
-   * @method createIndex
-   * @param {String} fileName
-   * @param {Object} jsonObject
+   * @param {Object} jsonObject the json data to build
    * @return {Object} index that was built
    */
-  createIndex(fileName, jsonObject) {
+  createIndex(jsonObject) {
     const index = {};
     jsonObject.map((sentence, count) => {
-      (`${sentence.title} ${sentence.text}`)
-        .replace(/[^A-Za-z0-9\s]/g, '')
-        .toLowerCase().split(' ')
-        .map((word) => {
-          if (index[word] && index[word].indexOf(count) === -1) {
-            index[word].push(count);
-          } else {
-            index[word] = [count];
-          }
-        });
+      this.filterString((`${sentence.title} ${sentence.text}`))
```

Comment:
Try to avoid using data types in your variable names (how about filterText instead of filterString).